File: 프로그래머스/lv1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    answer = ''
    d = {}
    for p in participant:
        d[p] = d.get(p,0) + 1
    for c in completion:
        d[c] -= 1
    for key, value in d.items():
        if value > 0:
            answer = key
    return answer

File: 프로그래머스/lv1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;
import java.util.Map.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        
        HashMap<String, Integer> hm = new HashMap<>();
        
        for(String p : participant) {
            hm.put(p, hm.getOrDefault(p, 0) + 1);
        }
        for(String c : completion) {
            hm.put(c, hm.get(c) - 1);
        }
        
        for (Entry<String, Integer> entrySet : hm.entrySet()) {
            if (entrySet.getValue() == 1) {
                answer = entrySet.getKey();
            }
        }
        return answer;
    }
}